src/System.Management.Automation/engine/remoting/commands/EnterPSHostProcessCommand.cs

```diff
@@ -565,14 +574,45 @@ internal static IReadOnlyCollection<PSHostProcessInfo> GetAppDomainNamesFromProc
                                     }
                                 }
 
-                                int pNameIndex = namedPipe.IndexOf(".", pAppDomainIndex + 1, StringComparison.OrdinalIgnoreCase);
+                                int pNameIndex = namedPipe.IndexOf('.', pAppDomainIndex + 1);
                                 if (pNameIndex > -1)
                                 {
                                     string appDomainName = namedPipe.Substring(pAppDomainIndex + 1, (pNameIndex - pAppDomainIndex - 1));
                                     string pName = namedPipe.Substring(pNameIndex + 1);
 
-                                    procAppDomainInfo.Add(
-                                        new PSHostProcessInfo(pName, id, appDomainName));
+                                    Process process = null;
+
+                                    // Only return instances that aren't ourselves since you cannot enter yourself
+                                    if (id != currentPid)
+                                    {
+                                        try
+                                        {
+                                            process = System.Diagnostics.Process.GetProcessById(id);
+                                        }
+                                        catch (Exception)
+                                        {
+                                            // Do nothing if the process no longer exists
+                                        }
+
+                                        if (process == null)
+                                        {
+                                            try
+                                            {
+                                                // If the process is gone, try removing the PSHost named pipe
+                                                var pipeFile = new FileInfo(namedPipe);
+                                                pipeFile.Delete();
+                                            }
+                                            catch (Exception)
+                                            {
+                                                // best effort to cleanup
+                                            }
+                                        }
+                                        else if (process.ProcessName.Equals(pName, StringComparison.Ordinal))
```

Comment:
I don't think this name check is necessary.  If the named pipe file is old or bogus then the process null check above should catch it.

Reply:
I think the intention is to cover the case that a process ID gets reused by another process.

Reply:
I'm thinking of the case where a pid gets reused by a different process

Reply:
But starttime currently is only decorative for uniqueness and not for validating if it's the same pwsh or a new one reusing the pid